source/RandomEvent.h

```diff
@@ -17,42 +17,74 @@ this program. If not, see <https://www.gnu.org/licenses/>.
 #define RANDOMEVENT_H_
 
 
+#include "RValue.h"
+
+
+
+template <class T>
+struct PeriodTypeHasConditions
+{
+	static bool HasConditions(const T &t) { return false; }
+};
+
+template <class V, class K>
+struct PeriodTypeHasConditions<RValue<V, K>>
+{
+	static bool HasConditions(const RValue<V, K> &t) { return t.WasLValue(); }
+};
+
 
 // A class representing an event that triggers randomly
 // in a given internal, for example fleets or hazards.
-template <typename T>
+template <typename T, typename P = int>
 class RandomEvent {
 public:
-	constexpr RandomEvent(const T *event, int period) noexcept;
+	typedef T EventType;
+	typedef P PeriodType;
+
+	constexpr RandomEvent(const T *event, const P &period) noexcept;
 
 	constexpr const T *Get() const noexcept;
-	constexpr int Period() const noexcept;
+	constexpr const P &Period() const noexcept;
+
+	// Update the period from a ConditionSet if needed:
+	template<typename Getter>
+	void UpdateConditions(const Getter &getter);
+
+	bool HasConditions() const { return PeriodTypeHasConditions<P>::HasConditions(period); }
 
 private:
 	const T *event;
-	int period;
+	P period;
```

Comment:
Why this change, it seems like you just want to use your RValue here wrapping and int.

Reply:
Yes. That should not have been an int. Thank you for catching that.